### PR TITLE
Multiple maintained backends in configuration

### LINKED ISSUE #14: Set more than one backend to maintenance mode

### PR DESCRIPTION
**Pull request by mjarco**
Resolves #14

### CODE REVIEW

File: dial/dial.go [hunk 1]
```diff
@@ -119,17 +119,22 @@ func (d *LimitDialer) Dial(network, addr string) (c net.Conn, err error) {
 	return c, err
 }
 
-// DropEndpoint marks backend as dropped i.e. maintenance x
-func (d *LimitDialer) DropEndpoint(endpoint url.URL) {
-	d.droppedEndpoint = endpoint.Host
+func (d *LimitDialer) setDropEndpoints(droppedEndpoints []url.URL) {
```
Comment (janisz) [inline dial/dial.go]: `s/setDropEndpoinst/setDroppedEndpoints` or `dropEndpoints`

File: dial/dial_test.go [hunk 2]
```diff
@@ -38,44 +41,44 @@ func TestLimitDialerMostLoadedEndpoint(t *testing.T) {
...
 	conn3, c3Err := dialer.Dial("tcp", addr2)
 	if !assert.Nil(t, conn3) {
 		defer func() {
 			err := conn3.Close()
-			assert.Nil(t, err)
+			assert.NoError(t, err)
 		}()
 	}
-	assert.NotNil(t, c3Err)
+	assert.Error(t, c3Err)
```
Comment (janisz) [inline dial/dial_test.go]: IMO this test need some comments by the asserts

File: Makefile [hunk 3]
```diff
@@ -12,6 +12,7 @@ lint: deps-lint
 	--vendor
 
 deps:
+	go get github.com/Masterminds/glide
```
Comment (janisz) [inline Makefile]: This do not belong to this feature. It's a bugfix and should be in master as soon as possible.

Comment (mjarco) [inline Makefile]: It's merge from master ;)

**Pull request merged**